Paws4Me/Paws4Me/Adoption Form/Adoption Form Page 1/View Controller/AdoptFirstFormPageViewController.swift

```diff
@@ -0,0 +1,204 @@
+            item.makeCircle()
+            item.addBorder()
+           }
+    }
+
+    // : MARK: - IBAction
+    @IBAction func form2NextButton(_ sender: Any) {
+
+        var indexTextfield = 0
+        while textFields.count != indexTextfield {
+            if let textField = self.textFieldForTag( tag: indexTextfield ) {
+                guard let textfieldValue = textField.text else { return }
+                var cellData = firstFormViewModel.arrayForm.filter({$0.cellIndex == indexTextfield})
+                cellData[0].data?.itemValue = textfieldValue
+                indexTextfield += 1
+            }
+    }
+
+        var indexSwitch = 0
+        while uiSwitches.count != indexSwitch {
+            if let uiSwitch = self.uiSwitchForTag(tag: indexSwitch) {
+                var cellData = firstFormViewModel.arrayForm.filter({$0.cellIndex == indexSwitch})
+                cellData[0].data?.itemValue = uiSwitch.isOn.description
+                indexSwitch += 1
+            }
+    }
+    }
+
+    // : MARK: - Functions
+
+    private  func textFieldForTag( tag: Int ) -> UITextField? {
+        return self.textFields.filter({ $0.tag == tag }).first
+    }
+
+    private  func uiSwitchForTag( tag: Int ) -> UISwitch? {
+        return self.uiSwitches.filter({ $0.tag == tag }).first
+    }
+
+   private func setUpTableView() {
+        adoptFirstFormTableView?.delegate = self
+        adoptFirstFormTableView?.dataSource = self
+        adoptFirstFormTableView?.register(TextBoxTableViewCell.self,
+                                          forCellReuseIdentifier: TextBoxTableViewCell.identifier)
+        adoptFirstFormTableView?.register(UISwitchTableViewCell.self,
+                                          forCellReuseIdentifier: UISwitchTableViewCell.identifier)
+        adoptFirstFormTableView?.reloadData()
+    }
+
+    // MARK: - UI Creation
+    func createTextField(question: String) -> UITextField {
+        let textField =  UITextField(frame: CGRect(x: 150, y: 10, width: 200, height: 40))
+        textField.placeholder = question
+        textField.font = UIFont.systemFont(ofSize: 15)
+        textField.borderStyle = UITextField.BorderStyle.roundedRect
+        textField.autocorrectionType = UITextAutocorrectionType.no
+        textField.keyboardType = UIKeyboardType.default
+        textField.returnKeyType = UIReturnKeyType.done
+        textField.clearButtonMode = UITextField.ViewMode.whileEditing
+        textField.contentVerticalAlignment = UIControl.ContentVerticalAlignment.center
+        textField.delegate = self as? UITextFieldDelegate
+        textField.tag = indexTextfield
+        self.textFields.append( textField )
+        indexTextfield += 1
+        return textField
+    }
+}
+
+// MARK: - UITableViewDataSource
+extension AdoptFirstFormPageViewController: UITableViewDataSource, UITableViewDelegate {
+    // MARK: - TableView functions
+     func numberOfSections(in tableView: UITableView) -> Int {
+        return 2
```

Comment:
Alright to leave it like this for now, but something to think about is how can we make this dynamic 😄

Reply:
Will do so 👍🏻